ush/python/pygfs/task/stat_analysis.py

```diff
@@ -0,0 +1,185 @@
+                'npz_anl': self.task_config.LEVS - 1,
+                'STAT_WINDOW_BEGIN': _window_begin,
+                'STAT_WINDOW_LENGTH': f"PT{self.task_config.assim_freq}H",
+                'OPREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
+                'APREFIX': f"{self.task_config.RUN}.t{self.task_config.cyc:02d}z.",
+                'GPREFIX': f"gdas.t{self.task_config.previous_cycle.hour:02d}z."
+            }
+        )
+        # Extend task_config with local_dict
+        self.task_config = AttrDict(**self.task_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self) -> None:
+        """
+        Initialize a global stat analysis
+        This method will initialize a global stat analysis.
+        This includes:
+        - initialize JEDI applications
+        - copying stat files
+        Parameters
+        ----------
+        None
+        Returns
+        ----------
+        None
+        """
+        # Create dictionary of Jedi objects
+        # Expected keys are what must be included from the JEDI config file. We can
+        # then loop through ob space list from scripts/exglobal_analysis_stats.py
+        expected_keys = ['aero', 'snow']
+        self.jedi_dict = Jedi.get_jedi_dict(self.task_config.JEDI_CONFIG_YAML, self.task_config, expected_keys)
+
+        logger.info(f"Copying files to {self.task_config.DATA}/stats")
+
+        # Loop through ob space list
+        for OB in self.task_config.STAT_OBS:
+            logger.info(f"Working on current observation: {OB}")
+
+            # Parse JEDI analysis stat jinja file
+            obs_dict = parse_j2yaml(self.task_config.JEDI_CONFIG_YAML, self.task_config)
+
+            # Copy stat files to DATA path
+            instat_files = os.path.join(obs_dict[OB]['stat_file_path'], f"{self.task_config['APREFIX']}{obs_dict[OB]['stat_file_name']}")
+            ob_dir_str = f"{self.task_config.DATA}" + f"/{OB}"
+            os.mkdir(ob_dir_str)
+
+            dest = os.path.join(ob_dir_str, obs_dict[OB]['stat_file_name'])
+            logger.info(f"Copying {instat_files} to {dest} ...")
+            statlist = [[instat_files, dest]]
+            FileHandler({'copy': statlist}).sync()
+
+            # Open tar file
+            logger.info(f"Open tarred stat file in {dest}")
+            with tarfile.open(dest, "r") as tar:
+                # Extract all files to the current directory
+                tar.extractall(path=f'{ob_dir_str}')
+
+            # Gunzip .nc files
+            logger.info("Gunzip files from tar file")
+            gz_files = glob.glob(os.path.join(ob_dir_str, "*.gz"))
+            logger.info(f"Gunzip files: {gz_files}")
+
+            for diagfile in gz_files:
+                output_file = os.path.join(ob_dir_str, os.path.basename(diagfile)[:-3])
+                with gzip.open(diagfile, 'rb') as f_in:
+                    with open(output_file, 'wb') as f_out:
+                        f_out.write(f_in.read())
```

Comment:
I believe I need to have a check after this to make sure there are files in the tar file. If not, move on to next ob. What are you thoughts?

Reply:
Yes, a check if it is non-empty is necessary